mne/preprocessing/artifact_detection.py

```diff
@@ -4,12 +4,89 @@
 
 
 import numpy as np
-
 from ..annotations import (Annotations, _annotations_starts_stops)
 from ..transforms import (quat_to_rot, _average_quats, _angle_between_quats,
                           apply_trans, _quat_to_affine)
+from ..filter import filter_data
 from .. import Transform
-from ..utils import (_mask_to_onsets_offsets, logger)
+from ..utils import (_mask_to_onsets_offsets, logger, verbose)
+
+
+@verbose
+def annotate_muscle(raw, threshold=1.5, picks=None, min_length_good=.1,
+                    filter_freq=[110, 140], n_jobs=1, verbose=None):
+    """Detect segments with muscle artifacts.
+
+    Detects segments periods that contain high frequency activity beyond the
+    specified threshold. Muscle artifacts are most notable in the range of
+    110-140 Hz.
+
+    Raw data is band-pass filtered between ``filter_freq`` especified
+    frequencies (default is 110 - 140 Hz), the signal envelope computed,
+    z-scored across samples, channel averaged and low-pass filtered to better
+    capture beginning and end of muscle activity and false positive transient
+    peaks.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to compute head position.
+    threshold : float
+        The threshold in z-scores for selecting segments with muscle activity
+        artifacts. Check ``scores_muscle`` to see optimal thesholding for the
+        data.
+    %(picks_all)s
+    min_length_good : int | float | None
+        The minimal good segment length between annotations, smaller segments
+        will be included in the movement annotation.
+    filter_freq : list | tuple, default (110, 140)
+        The lower and upper high frequency to filter the signal for muscle
+        detection.
+    %(n_jobs)s
+    %(verbose)s
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with muscle artifacts annotated as BAD_muscle.
+    scores_muscle : array
+        Z-score values averaged across channels for each sample.
+    """
+    from scipy.stats import zscore
+    from scipy.ndimage.measurements import label
+
+    raw_copy = raw.copy()
+    raw_copy.pick(picks)
+    ch_type = raw_copy.get_channel_types()
+
+    # Remove ref chans if MEG data just in case
+    meg = [True for e in ['mag', 'grad'] if (e in ch_type)]
+    if meg is True:
+        raw_copy.pick_types(ref_meg=False)
+
+    # Only one type of channel, otherwise z-score will be biased
+    assert(len(set(ch_type)) == 1), 'Different channel types, pick one type'
+
+    raw_copy.filter(filter_freq[0], filter_freq[1], fir_design='firwin',
+                    pad="reflect_limited", n_jobs=n_jobs)
+    raw_copy.apply_hilbert(envelope=True, n_jobs=n_jobs)
+    sfreq = raw_copy.info['sfreq']
+
+    art_scores = zscore(raw_copy.get_data(reject_by_annotation="NaN"), axis=1)
+    scores_muscle = filter_data(art_scores.mean(axis=0), sfreq, None, 4)
```

Comment:
This should be `art_scores.sum(axis=0) / np.sqrt(n_channels)` to follow what fieldtrip does.

Reply:
You should be consistent, for some things you want to follow FT others your own 🤔

Reply:
1. I think this calculation makes more sense than simple average - when you have lots of channels it should still be sensitive to artifacts on a smaller subset of channels
2. Following FT in this regard will also allow to more easily compare thresholds between mne and FT - and for example reproduce muscle rejection reported in a paper when a threshold is given.